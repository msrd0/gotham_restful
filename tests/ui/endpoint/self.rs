#[macro_use]
extern crate gotham_restful;

#[derive(Resource)]
#[resource(read_all)]
struct FooResource;

#[read_all]
fn read_all(self) {}

fn main() {}
