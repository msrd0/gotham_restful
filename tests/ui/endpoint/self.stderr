error: Didn't expect self parameter
 --> tests/ui/endpoint/self.rs:9:13
  |
9 | fn read_all(self) {}
  |             ^^^^

error: `self` parameter is only allowed in associated functions
 --> tests/ui/endpoint/self.rs:9:13
  |
9 | fn read_all(self) {}
  |             ^^^^ not semantically valid as function parameter
  |
  = note: associated functions are those in `impl` or `trait` definitions

error[E0412]: cannot find type `read_all___gotham_restful_endpoint` in this scope
 --> tests/ui/endpoint/self.rs:5:12
  |
5 | #[resource(read_all)]
  |            ^^^^^^^^ not found in this scope
